homeassistant/core.py

```diff
@@ -1749,7 +1749,9 @@ class Service:
 
     def __init__(
         self,
-        func: Callable[[ServiceCall], Coroutine[Any, Any, ServiceResponse] | None],
+        func: Callable[
+            [ServiceCall], Coroutine[Any, Any, ServiceResponse] | ServiceResponse | None
```

Comment:
I don't think we should change core in this PR. Please make a separate PR for that.

Reply:
Done in https://github.com/home-assistant/core/pull/102813